### ISSUE TITLE
https://semgrep.dev/s/xeLl should report 1 fail but it reports 2

### ISSUE THREAD

**Issue by raghavjain3** [semgrep/semgrep#3544]
**Describe the bug**
https://semgrep.dev/s/xeLl should report 1 fail but it reports 2

**Expected behavior**
Only 1 test should fail

**Screenshots**
![image](https://user-images.githubusercontent.com/32291067/126009838-4cc8c2fe-e444-461d-9c2b-035914f3cccf.png)

**What is the priority of the bug to you?**
P1

**Environment**
semgrep.dev

**Comment by tilperion2** [comment]
Found the same in my automation project. Some rules get duplicated output. 
So my test cases are failing. I think problem appears only in recent semGrep updates > 0.58.x

Example: https://semgrep.dev/s/Gwz1/

PS. not sure though, that is is really connected with the issue in the top comment.

**Comment by guifre** [comment]
+1 Seeing Impact here as well, we have a broken pipeline due to this.

**Comment by tilperion2** [comment]
> Found the same in my automation project. Some rules get duplicated output.
> So my test cases are failing. I think problem appears only in recent semGrep updates > 0.58.x
> 
> Example: https://semgrep.dev/s/Gwz1/
> 
> PS. not sure though, that is is really connected with the issue in the top comment.

I`ve tried to revert to different previous versions to unblock our build pipeline. 
Found that latest semGrep that does not duplicate finding in result is v0.56.0.
Hope it may help.

**Comment by emjin** [comment]
@aryx Could https://github.com/returntocorp/semgrep/pull/3310 be related? It was merged in v0.57.0.

**Comment by IagoAbal** [comment]
Are we sure this is a problem with Semgrep and not with Semgrep App? https://semgrep.dev/s/xeLl?version=0.56.0 does not work any better, and neither does https://semgrep.dev/s/xeLl?version=0.55.0.

**Comment by tilperion2** [comment]
Hi, not sure about top one, but mine example shows the difference:
https://semgrep.dev/s/Gwz1?version=0.60.0 - two similar/duplicated finding for the same line
https://semgrep.dev/s/Gwz1?version=0.56.0 - only one finding reported - correct one.

**Comment by IagoAbal** [comment]
Thanks for pointing it out @tilperion2, I think these are two different issues. Yours is probably a Semgrep issue, will take a look.

**Comment by IagoAbal** [comment]
@tilperion2 could you open a separate issue?

**Comment by IagoAbal** [comment]
@raghavjain3 I believe this issue should be reported to Semgrep-app. Semgrep is not doing anything suspicious as far as I can tell, it might be a bug in the `evaluate_tests` function which returns the number of failed tests.

**Comment by raghavjain3** [comment]
> @raghavjain3 I believe this issue should be reported to Semgrep-app. Semgrep is not doing anything suspicious as far as I can tell, it might be a bug in the `evaluate_tests` function which returns the number of failed tests.

I will leave it open here and open another issue in semgrep-app but please feel free to close this out if you don't think it's relevant.

**Comment by nbrahms** [comment]
Closing in favor of returntocorp/semgrep-app#3041
